packages/common/src/configuration/config-manager.ts

```diff
@@ -143,18 +151,169 @@ class PochiConfigManager {
     await this.save();
   };
 
-  get config(): ReadonlySignal<PochiConfig> {
+  get config(): Signal<PochiConfig> {
     return this.cfg;
   }
 }
 
-const { config, updateConfig, getVendorConfig, updateVendorConfig } =
-  new PochiConfigManager();
+class PochiConfigManager {
+  private userConfigFile: PochiConfigFile;
+  private workspaceConfigFile: PochiConfigFile | null = null;
+
+  private readonly mergedConfig: Signal<PochiConfig> = signal({});
+
+  constructor() {
+    this.userConfigFile = new PochiConfigFile(UserConfigFilePath);
+    if (process.env.POCHI_SESSION_TOKEN) {
+      this.userConfigFile.config.value = mergeDeep(
+        this.userConfigFile.config.value,
+        {
+          vendors: {
+            pochi: {
+              credentials: {
+                token: process.env.POCHI_SESSION_TOKEN,
+              },
+            },
+          },
+        },
+      );
+    }
+    this.userConfigFile.config.subscribe(this.updateMergedConfig);
+    this.updateMergedConfig();
+  }
+
+  private updateMergedConfig = () => {
+    // current only allow workspace to override mcp setting
+    const mergeKeys = ["mcp"] as const;
```

Comment:
make this a global variable within this file to make it explicit, e.g `AllowedWorkspaceConfigKeys`